cli/loader/src/lib.rs

```diff
@@ -347,7 +349,11 @@ impl Loader {
         parser_path: &Path,
         scanner_path: &Option<PathBuf>,
     ) -> Result<Language> {
-        let mut library_path = self.parser_lib_path.join(name);
+        let mut lib_name = name.to_string();
+        if self.debug_build {
+            lib_name.push_str(".debug._");
```

Comment:
This is used to generate separate debug parser library versions like `~/.cache/tree-sitter/lib/javascript.debug.so` to don't affect regular versions when a user calls the `tree-sitter parse` without the `--debug-build` flag.